packages/core/src/browser/styling-service.ts

```diff
@@ -53,11 +53,24 @@ export class StylingService implements FrontendApplicationContribution {
     protected readonly themingParticipants: ContributionProvider<StylingParticipant>;
 
     onStart(): void {
-        this.applyStyling(this.themeService.getCurrentTheme());
-        this.themeService.onDidColorThemeChange(e => this.applyStyling(e.newTheme));
+        this.registerWindow(window);
+        this.themeService.onDidColorThemeChange(e => this.applyStylingToWindows(e.newTheme));
+    }
+
+    registerWindow(win: Window): Disposable {
+        const cssElement = DecorationStyle.createStyleElement('contributedColorTheme', win.document.head);
+        this.cssElements.set(win, cssElement);
+        this.applyStyling(this.themeService.getCurrentTheme(), cssElement);
+        return Disposable.create(() => this.cssElements.delete(win));
+    }
+
+    protected applyStylingToWindows(theme: Theme): void {
+        this.cssElements.forEach((cssElement, win) => {
+            this.applyStyling(theme, cssElement);
+        });
```

Comment:
Suggestion: `win` seems to be unused.
```suggestion
        this.cssElements.forEach(cssElement => {
            this.applyStyling(theme, cssElement);
        });
```